corehq/apps/locations/models.py

```diff
@@ -678,15 +678,16 @@ def save(self, *args, **kwargs):
             ]
             self.site_code = generate_code(self.name, all_codes)
 
+        # Set the UUID here so we can save to SQL first (easier to rollback)
+        if not self.location_id:
+            self._id = self.get_db().server.next_uuid()
```

Comment:
`self.location_id` is just a property that returns `self._id` - I do agree that we should check `if not self._id` for clarity though - will update.